File: src/Lesson5/java/Ex7.java
package Lesson5.java;
import java.util.Arrays;

public class Ex7 {

    public static void moveZerosToEnd(int[] arr) {

        int index = 0;

        for (int num : arr) {

            if (num != 0) {
                arr[index] = num;
                index++;
            }
        }

        while (index < arr.length) {
            arr[index] = 0;
            index++;
        }
    }

    public static void main(String[] args) {

        int[] arr = {0, 1, 0, 3, 12};

        System.out.println("Original array: " + Arrays.toString(arr));

        moveZerosToEnd(arr);

        System.out.println("Modified array: " + Arrays.toString(arr));
    }
}

File: src/Lesson5/forkotlin/Ex7.kt
package Lesson5.forkotlin

fun moveZerosToEnd(arr: IntArray) {

    var index = 0

    for (num in arr) {

        if (num != 0) {
            arr[index] = num
            index++
        }
    }

    while (index < arr.size) {
        arr[index] = 0
        index++
    }
}

fun main() {

    val arr = intArrayOf(0, 1, 0, 3, 12)

    println("Original array: ${arr.joinToString()}")

    moveZerosToEnd(arr)

    println("Modified array: ${arr.joinToString()}")
}